webapp/src/compiler.ts

```diff
@@ -431,10 +430,16 @@ export function formatAsync(input: string, pos: number) {
 }
 
 export function snippetAsync(qName: string, python?: boolean): Promise<string> {
-    return workerOpAsync("snippet", {
+    let initStep = Promise.resolve();
+    if (python) {
+        // To make sure that the service is working with the most recent version of the file,
+        // run a typecheck
+        initStep = typecheckAsync().then(() => {})
```

Comment:
This seems like it should be handled by the language service. It can itself recompile things if need be. Is there a reason it has to be done here?

Reply:
we don't send the whole project to the language service with each request for snippets. i was seeing a bug where sometimes the version of the file in the language service did not match the version in the editor; calling a typecheck forces the webapp to send over all the files.

Reply:
I see. It'd be great if there was an explicit "syncFiles", or even better would be if we had a version notion and when the webapp is calling out to the worker they could check last versions and ensure files are synced. But this is likely out of scope.